### PR TITLE
Make better config

### PR DESCRIPTION
**Pull request by oanhnn**
- [x] Make better `docker-compose` config with `networks` instead `links`
- [x] Make better `nginx` and `php-fpm` config
- [x] Simpler fix permission for writable directories in volumes
- [x] Add more test: upload, connect database, writable directories, ...

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -33,8 +35,9 @@ services:
       - ./php7.0-fpm/php-fpm.conf:/usr/local/etc/php-fpm.conf
       - ./storage/logs:/var/log/php-fpm
       - .:/app
-    links:
-      - db
+    networks:
+      - frontend
```
Comment (hnq90) [inline docker-compose.yml]: @oanhnn I think the network should be `internal`/`external` or `private`/`public`.
How's about you?

File: php7.0-fpm/Dockerfile [hunk 2]
```diff
@@ -8,14 +8,15 @@ RUN apt-get update && apt-get install -y \
...
-        npm \
         git \
+        build-essential \
     && docker-php-ext-install -j$(nproc) iconv mcrypt pdo_mysql intl opcache mbstring curl zip \
     && docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ \
     && docker-php-ext-install -j$(nproc) gd
 
 RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
-RUN apt-get clean
+RUN curl -sL https://deb.nodesource.com/setup_6.x | bash -
```
Comment (hnq90) [inline php7.0-fpm/Dockerfile]: @oanhnn I think we should move this line to the top of Dockerfile.
So we can remove the next line (`RUN apt-get install -y nodejs && apt-get autoremove && apt-get clean`)  and add `nodejs` after `build-essential`.

So our image will be reduced 1 layer than before.

**Review by hnq90 — COMMENTED** [review]
Please check my review :grin:

File: docker-compose.yml (hunk 1 above)
Comment (oanhnn) [reply docker-compose.yml]: I think ok

File: php7.0-fpm/Dockerfile (hunk 2 above)
Comment (oanhnn) [reply php7.0-fpm/Dockerfile]: It is so cool!

Comment (oanhnn) [reply php7.0-fpm/Dockerfile]: But i think we shouldn't remove line
```
RUN apt-get autoremove && apt-get clean
```

Comment (hnq90) [reply php7.0-fpm/Dockerfile]: @oanhnn Yes, we can chain those commands after install `nodejs`.

```
build-essential \
nodejs \
&& apt-get autoremove \
&& apt-get clean \
&& docker-php-ext-install -j$(nproc) iconv mcrypt pdo_mysql intl opcache mbstring curl zip \
```

**Review by hnq90 — APPROVED** [review]
LGTM :+1:

File: docker-compose.yml [hunk 3]
```diff
@@ -43,15 +46,20 @@ services:
...
       - .:/app
-    links:
-      - php7
+    networks:
+      - external
     dns:
       - 8.8.8.8
       - 8.8.4.4
+
+networks:
```
Comment (hnq90) [inline docker-compose.yml:61]: @oanhnn OMG, you forgot to change the network aliases here...

**Pull request merged**